kybra/compiler/kybra_generate/src/canister_method/pre_upgrade_method/build.rs

```diff
@@ -16,12 +16,12 @@ use crate::{
 impl PyAst {
     pub fn build_pre_upgrade_method(&self) -> Result<Option<PreUpgradeMethod>, Vec<Error>> {
         let pre_upgrade_function_defs =
-            self.get_canister_stmt_of_type(CanisterMethodType::PreUpgrade);
+            self.get_canister_stmt_of_type(&CanisterMethodType::PreUpgrade);
```

Comment:
The reason this was added was because of introducting the init_or_post_upgrade version that passed the CansiterMethodType as a parameter right? In the previous version everyone was isolated to their own file so you could just make a new instance of that enum and not pass in references. Is that correct?

If so then I think it makes a lot more sense to look into implementing the copy trait

Reply:
Copy implemented and all references removed